text/0060-refashion-updates-apis.md

```diff
@@ -0,0 +1,509 @@
+- 🔴 If a master key is sent by the client but does not match the value configured on the server-side, the API returns a `403 Forbidden` - `invalid_token`.
+- 🔴 If the index does not exist, the API returns a `404 Not Found` - `index_not_found` error.
+- 🔴 If the task does not exist, the API returns a `404 Not Found` - `task_not_found` error.
+
+#### 6. `task_not_found` error
+
+##### Context
+
+This error happens when a requested task can't be found.
+
+##### Error Definition
+
+HTTP Code: `404 Not Found`
+
+```json
+{
+    "message": "Task :taskUid not found.",
+    "code": "task_not_found",
+    "type": "invalid_request",
+    "link": "https://docs.meilisearch.com/errors#task_not_found"
+}
+```
+
+- The `:taskUid` is inferred when the message is generated.
+
+#### 7. `MEILI_MAX_UDB_SIZE` env var and `--max-udb-size` cli option
+
+As the notion of `update` no longer exists. The acronym `UDB` is changed by `TASK_DB`.
+
+- `MEILI_MAX_UDB_SIZE` env var is updated to `MEILI_MAX_TASK_DB_SIZE`.
+- `--max-udb-size` cli option is updated to `--max-task-db-size`.
+
+#### 8. Asynchronous Write Operations on Index resource
+
+To consolidate the writing behavior between an index and document modifications, configuration changes of settings, the creation, modification, and deletion of an index resource are now asynchronous.
+
+Initially, the index creation, update, and deletion operations were synchronous, which could cause problems like race conditions.
+
+For example, when updating documents on an index while an index is being deleted synchronously in parallel. It also improves the consistency for the understanding of writes within a MeiliSearch index among an identical behavior.
+
+This structure allows us to facilitate communications and write propagations in a high availability context in the future.
+
+The main change in the API is that the routes response described below now becomes a `202 Accepted` response with the summarized task response payload.
+
+Errors are now part of the `task` as for other asynchronous operations.
+
+New task types are also added for these operations. `indexCreation`, `indexUpdate`, and `indexDeletion`.
+
+**Create an index** | `POST` - `/indexes`
+
+```json
+{
+    "uid": "movies"
+}
+```
+
+`202` - Accepted Response body
+
+```json
+{
+    "uid": 0,
+    "indexUid": "movies",
+    "status": "enqueued",
+    "type": "indexCreation",
+    "enqueuedAt": "2021-08-12T10:00:00.000000Z"
+}
+```
+
+- 💡 Automatic index creation using the `/indexes/:indexToCreate/documents` route generates only one `documentAdditions` task that also handles index creation.
+
+**Update an index** | `PUT` - `/indexes/:indexUid`
+
+```json
+{
+    "primaryKey": "uid"
+}
+```
+
+`202` - Accepted Response body
+
+```json
+{
+    "uid": 1,
+    "indexUid": "movies",
+    "status": "enqueued",
+    "type": "indexUpdate",
+    "enqueuedAt": "2021-08-12T10:00:00.000000Z"
+}
+```
```

Comment:
Is this response the expected response for any asynchronous task (ex: add documents, add settings)? Sorry if it's already in the spec, I'm a little bit lost since they are a lot of changes, maybe it's visible in open API

Reply:
Yes, this is a summarized `task` response only for the `202 Accepted`. The goal is to give more details instead of just an identifier as a response.